src/org/zaproxy/zap/extension/pscanrulesAlpha/ServerHeaderInfoLeakScanner.java

```diff
@@ -61,29 +67,28 @@ public void scanHttpRequestSend(HttpMessage msg, int id) {
 	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 		long start = System.currentTimeMillis();
 	
-		Vector<String> serverOption = msg.getResponseHeader().getHeaders("Server");
-		if (serverOption != null) { //Header Found
-			//It is set so lets check it. Should only be one but it's a vector so iterate to be sure.
-			for (String serverDirective : serverOption) {
-				boolean matched = VERSION_PATTERN.matcher(serverDirective).matches();
-				if (matched) { //See if there's any version info.
-					//While an alpha string might be the server type (Apache, Netscape, IIS, etc) 
-					//that's much less of a head-start than actual version details.
-					Alert alert = new Alert(getPluginId(), Alert.RISK_LOW, Alert.CONFIDENCE_MEDIUM, //PluginID, Risk, Reliability
-						getName()); 
-		    			alert.setDetail(
-		    					getDescription(), //Description
-		    					msg.getRequestHeader().getURI().toString(), //URI
-		    					"",	// Param
-		    					"", // Attack
-		    					"", // Other info
-		    					getSolution(), //Solution
-		    					getReference(), //References
-		    					serverDirective,	// Evidence - Return the Server Header info
-		    					200, // CWE Id 
-		    					13,	// WASC Id 
-		    					msg); //HttpMessage
-		    		parent.raiseAlert(id, alert);
+		Vector<String> serverOption = msg.getResponseHeader().getHeaders("Server");	
+		if (serverOption != null) { //Header Found			
+			if (Plugin.AlertThreshold.LOW.equals(this.getLevel())) {
+				// If we are operating with an LOW treshhold, detecting a
+				// "server" header is enough to raise an alert.
+				final Alert alert = buildAlert(AlertType.SERVER, Alert.RISK_INFO, serverOption.toString(), msg);

```

Comment:
`serverOption.toString()` will only successfully highlight in the response if the multiple headers happen to be contiguous, which is probably not an assumption that should be made.

Also wouldn't we still want to iterate them and see if the version is in fact disclosed in one of them?